sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2858,15 +2858,15 @@ object SQLConf {
       .createWithDefault(true)
 
   val USE_NULLS_FOR_MISSING_DEFAULT_COLUMN_VALUES =
-    buildConf("spark.sql.defaultColumn.useNullsForMissingDefautValues")
+    buildConf("spark.sql.defaultColumn.useNullsForMissingDefaultValues")
       .internal()
       .doc("When true, and DEFAULT columns are enabled, allow column definitions lacking " +
         "explicit default values to behave as if they had specified DEFAULT NULL instead. " +
         "For example, this allows most INSERT INTO statements to specify only a prefix of the " +
         "columns in the target table, and the remaining columns will receive NULL values.")
       .version("3.4.0")
       .booleanConf
-      .createWithDefault(false)
+      .createWithDefault(true)
```

Comment:
This is a breaking change. Why do we have to make it `true`? (Probably I miss some discussions in this PR.)

Reply:
@gengliangwang No problem, per our discussion:
* The previous behavior when inserting fewer values than the number of columns in the table was to return an error, e.g. `CREATE TABLE t (a INT, b INT); INSERT INTO t VALUES 42'.
* When updating this `USE_NULLS_FOR_MISSING_DEFAULT_COLUMN_VALUES` config to true, then such commands would instead succeed and insert NULL values for remaining columns in the table. This is the behavior that some other engines use, e.g. Postgres, which some users have requested in Spark.
* This would be a behavior change (which some users may not expect) but not technically a breaking change because it changes only commands that would have returned errors to completing successfully instead.

![image](https://user-images.githubusercontent.com/99207096/164286719-f51d5dd0-05da-4aff-bf2a-f028355c6b3a.png)

Reply:
Maybe we could leave the config as `false` in this PR to not block merging it, and discuss the change with others for a few days? Then if we decide to change it to true, we can do so in a separate PR, WDYT?

Reply:
Update on this: we talked a bit and decided it would be best to keep this config false by default. We can retain the test coverage but leave the existing OSS Spark behavior as-is. Thanks for discussing the details for this.

Reply:
OK, I'll change it back to false.

Reply:
Done. Code updated, build&test succeeded